src/Microsoft.AspNet.Mvc.ModelBinding/IModelMetadataProvider.cs

```diff
@@ -13,7 +13,5 @@ public interface IModelMetadataProvider
         ModelMetadata GetMetadataForType([NotNull] Type modelType);
 
         IEnumerable<ModelMetadata> GetMetadataForProperties([NotNull] Type modelType);
-
-        ModelMetadata GetMetadataForParameter([NotNull] ParameterInfo parameter, IEnumerable<object> attributes);
```

Comment:
Also remove `MetadataAttributes.GetAttributesForParameter` and `MetadataKind.Parameter`.